core/arch/x86/encode.c

```diff
@@ -2589,10 +2643,11 @@ instr_encode_arch(dcontext_t *dcontext, instr_t *instr, byte *copy_pc, byte *fin
     }
 
     /* second opcode byte, if there is one */
-    if (TEST(OPCODE_TWOBYTES, info->opcode)) {
+    if (TEST(OPCODE_TWOBYTES, info->opcode) || TEST(OPCODE_THREEBYTES, info->opcode)) {
```

Comment:
This does not look avx-512 specific -- are you sure this is needed?  Wouldn't this mean sthg was broken before?

Reply:
That was a bug IMO. But apparently no instructions needed it. The encoder encode the primary opcode for three byte versions if that's missing. I think there may not have been any three-byte VEX opcodes until now. I didn't see any when I checked.

Reply:
It looks like OPCODE_TWOBYTES is always set when OPCODE_THREEBYTES is set so I do not think there is any bug here.  Instead, the decode_private.h docs should make that explicit.